File: P1.cpp
#include <iostream>
using namespace std;

// Sum of all the multiples of 3 or 5 below 1000

int sum(int n){
   if(n == 0) return 0;
   if((n%3 == 0) || (n%5 == 0)) return sum(n-1) + n;
   return sum(n-1);
}

int main(){
   int x = sum(999);
   cout << x << endl;
}

File: project-euler/python/p1.py
#!/usr/bin/env python3
"""
P1: Multiples of 3 and 5
"""


def p1(maxN):
    return sum(x for x in range(maxN) if not x % 3 or not x % 5)


print("Max is 10:", p1(10))  # 23
print("Max is 1000:", p1(1000))  # 233168
